internal/domain/usecases/user_usecase.go

```diff
@@ -31,8 +31,28 @@ func NewUserUsecase(cfg config.Config, logger *zap.Logger, userRepository reposi
 
 // super-admin method
 
-func (u *userUsecase) GetAllUsers(req *dtos.UserDTO) (*[]dtos.UserDTO, *apperror.AppError) {
-	return nil, nil
+func (u *userUsecase) GetAllUsers(req *dtos.GetAllUsersDTO) (*[]dtos.UserDTO, *apperror.AppError) {
+	offset := req.Limit * (req.Page - 1)
+	limit := req.Limit
+
+	users, err := u.userRepository.FindAllUsers(limit, offset)
+	if err != nil {
+		u.logger.Named("GetAllUsers").Error(constant.ErrUserNotFound, zap.Error(err))
+		return nil, apperror.BadRequestError(constant.ErrUserNotFound)
```

Comment:
imho, let's use internal server error instead bc we would've checked for bad request errors in handlers layer already. So if an error's gonna occur in repo, it's more likely to be 500 i think